monkey/infection_monkey/transport/tcp.py

```diff
@@ -48,9 +63,26 @@ def run(self):
 
         self.source.close()
         self.dest.close()
+        if self._client_disconnected:
+            self._client_disconnected()
 
 
 class TcpProxy(TransportProxyBase):
```

Comment:
Why do we need to inherit from `TransportProxyBase`?

For these classes, I'd prefer we just create new ones that suit our needs and leave the old ones in place, especially since we're creating small PRs and merging to `develop`. These changes will break the existing proxy, which needs to remain "working" until we're ready to cut over to the relay.